File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/AtEndOfTurnMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import AbstractCreature from "../Unit/AbstractCreature";

export class AtEndOfTurnMessage implements IMessage{
    public static readonly Id = "AtEndOfTurnMessage"; 
    public constructor(public IsPlayer:boolean,public unit:AbstractCreature) {
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/AtEndOfTurnMessage.cs
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///在回合结束
    public class AtEndOfTurnMessage : IEventHandle
    {
        public bool IsPlayer;
        public AbstractCreature Unit;
        public AtEndOfTurnMessage(bool isPlayer, AbstractCreature unit)
        {
            IsPlayer = isPlayer;
            Unit = unit;
        }
    }
}